### PR TITLE
Impliment Sprite Composition: Co-Authored by Liam and Dan

### PR DESCRIPTION
**Pull request by Thorakii**

### CODE REVIEW

**Comment by Thorakii** [comment]
[Add Sprite Composition Logic](https://trello.com/c/YseBmUbz/95-add-sprite-composition-logic)

File: src/bullet.py [hunk 1]
```diff
@@ -1,43 +1,72 @@
...
+        sprite.owner = self
+        sprite_list.append(self.sprite)
+        self.dps = 0
+        self.weapon = weapon
+
 
 # For naming purposes, a bullet can be anything that comes out of a weapon included beams, rockets, etc
 def bullet_behavior(
-    delta_time, player_manager: PlayerManager, list_of_walls: arcade.SpriteList
+    delta_time,
```
[the remark this answers is not in the document]
Comment (Thorakii) [reply src/bullet.py:41]: Fixed

File: src/bullet.py [hunk 2]
```diff
@@ -1,43 +1,72 @@
...
+    player_spritelist: arcade.SpriteList,
```
[the remark this answers is not in the document]
Comment (Thorakii) [reply src/bullet.py:42]: Fixed

File: src/player.py [hunk 3]
```diff
@@ -1,14 +1,22 @@
...
 class Player:
-    def __init__(self, input: PlayerInput):
-        self.sprite: SpriteForPlayer = SpriteForPlayer(self)
+    def __init__(
+        self,
+        input: PlayerInput,
+        projectile_spritelist: arcade.SpriteList,
+        beam_spritelist: arcade.SpriteList,
+    ):
+        # self.sprite: SpriteForPlayer = SpriteForPlayer(self)
```
[the remark this answers is not in the document]
Comment (Thorakii) [reply src/player.py:17]: Fixed

File: src/weapon.py [hunk 4]
```diff
@@ -55,76 +64,76 @@ class LaserBeam(Weapon):
...
+        self.muzzle_transform = (20 + self.beam_range / 2, 5, 0)
+        self.my_beam_sprite = None
 
-    def update(
-        self,
-        delta_time,
-        projectile_list: arcade.SpriteList,
-        beam_list: arcade.SpriteList,
-    ):
+    def update(self, delta_time):
```
[the remark this answers is not in the document]
Comment (Thorakii) [reply src/weapon.py:79]: Fixed

File: src/weapon.py [hunk 5]
```diff
@@ -55,76 +64,76 @@ class LaserBeam(Weapon):
...
+
+    def swap_out(self):
+        self.remove_beam()
 
-    def swap_out(self, beam_list: arcade.SpriteList):
-        if self.beam_projection in beam_list:
-            beam_list.remove(self.beam_projection)
+    def remove_beam(self):
+        if self.my_beam_sprite in self.player.beam_spritelist:
+            self.player.beam_spritelist.remove(self.my_beam_sprite)
```
[the remark this answers is not in the document]
Comment (Thorakii) [reply src/weapon.py:99]: In a next branch I'm going to play with who has responsibility for moving the beam around and see what seems to feel best

File: src/main.py [hunk 6]
```diff
@@ -34,11 +34,13 @@ def __init__(self, width, height, title):
 
     def setup(self):
         self.all_sprites = arcade.SpriteList()
-
+        self.projectile_spritelist = arcade.SpriteList()
```
[the remark this answers is not in the document]
Comment (Thorakii) [reply src/main.py:37]: Fixed

File: src/weapon.py [hunk 7]
```diff
@@ -55,76 +64,76 @@ class LaserBeam(Weapon):
...
         # the -45 degree angle in the offset corrects for rocket texture angled up 45 degrees
         self.muzzle_transform = (30, 2, math.radians(-45))
 
-    def update(
-        self,
-        delta_time,
-        projectile_list: arcade.SpriteList,
-        beam_list: arcade.SpriteList,
-    ):
+    def update(self, delta_time):
```
[the remark this answers is not in the document]
Comment (Thorakii) [reply src/weapon.py:119]: Fixed

**Review by cspotcode — APPROVED** [review]

**Pull request merged**